src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -254,42 +256,49 @@ public async Task ProcessFinishedReleasesAsync(CancellationToken cancellationTok
                                 else
                                 {
                                     Logger.LogInformation($"Release {releaseId}, channel {channelId} finished executing");
+
+                                    if (release.Environments[0].Status == AzureDevOpsReleaseStatus.Rejected)
```

Comment:
We'll need to check other statuses here. Any of these status means a problem and the release won't proceed/succeed: undefined, canceled, rejected, partiallySucceeded

Reply:
Also, would be better to perform the check on all environments, like what is done on line 251